wapitiCore/net/crawler.py

```diff
@@ -1016,6 +1017,93 @@ def __init__(
         self._auth_credentials = {}
         self._auth_method = "basic"
 
+    def get_auth_form(self, url):
+        head = {
+            'User-Agent': self.user_agent,
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
+            'Accept-Encoding': 'gzip, deflate, br'
+        }
+
+        if len(self._auth_credentials) == 2:
+
+            list = ['email', 'user', 'mail', 'session', 'login']
+            username, password = self._auth_credentials
+            response = requests.get(url, allow_redirects=True, verify=False, headers=head)
+            req= web.Request(url)
+            page = self.get(req, follow_redirects=True, headers=head)
+            forms = []
+            for i, forma in enumerate(page.iter_forms(autofill=False)):
+                forms.append(forma)
+            form = forms[0]
+            post_params = form.post_params
+            for i, post_param_tuple in enumerate(post_params):
+                field, value = post_param_tuple
+                if value:
+                    post_params[i] = [field, value]
+                elif "password" in field:
+                    post_params[i] = [field, password]
+                elif any(ext in field for ext in list):
+                    post_params[i] = [field, username]
+
+            if form.file_name:
+                request = Request(form.url, post_params=post_params, link_depth=form.link_depth)
+            else:
+                request = Request(page._response.url, post_params=post_params, link_depth=form.link_depth)
+
+            response = self.post(request, follow_redirects=True,headers=head)
+
+        else:
+
+            page = self.get(web.Request(url), follow_redirects=True, headers=head, verify=False)
+            forms = []
+
+            for i, form in enumerate(page.iter_forms(autofill=False)):
+                if i == 0:
+                    print('')
+                    print(_("Choose the form you want to use or enter 'q' to leave :"))
```

Comment:
This part works well as we are asking user input but see my comment on `set_session`